### PR TITLE
Modus Table - Add support for Expandable rows 

### PR DESCRIPTION
**Pull request by Ankur-Koul**
## Description
Allows users to add data as sub/child rows and view it by expanding the parent row.

References #1114 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
Able to hide first column 
image

**Comment by Ankur-Koul** [comment]
> * Please increase the number of rows for the expandable row page,
Now there are 7 rows in total.
> * I feel we need a better name for nested rows, in place of `subRows`. Simply naming it `rows` sounds better.
`subRows` is a property of `tanstack`, we can not change this.
> * Show at least 4 levels
Done

**Comment by Ankur-Koul** [comment]
> 1. Please increase the number of rows for the expandable row page,
> 2. I feel we need a better name for nested rows, in place of `subRows`. Simply naming it `rows` sounds better.
> 3. Show at least 4 levels

Hi

**Comment by HarshithaNous** [comment]
Below are my observations:
**Scenario 1:**
When the page view is set to 2 or 5, and the parent row is expanded the child rows are moved to next page. We expect the child rows to be displayed along with the parent row. Please refer the below screenshots:
image
image

**Scenario 2:**
Showing results displays an incorrect count as "Showing result1-20 of 7". The Total count should be more than the results displayed per page
image

**Comment by HarshithaNous** [comment]
There is no color code specified to verify the color of the expandable rows

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 1]
```diff
@@ -108,6 +117,20 @@ export class ModusTable {
     this.table.options.state.columnOrder = this.columnOrder;
   }
 
+  /** (Optional) To display expanded rows. */
+  @Prop() isExpand = false;
+  @Watch('isExpand') onChangeOfIsExpand() {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-table/modus-table.tsx:122]: I suggest keeping the option name `rowsExpandable`, as it is clear and easy to understand.

File: stencil-workspace/src/components/modus-table/utilities/drag-and-drop.utility.tsx [hunk 2]
```diff
@@ -7,26 +7,27 @@ export class DragAndDrop {
   columnReorder = false;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-table/utilities/drag-and-drop.utility.tsx:7]: I would suggest renaming the class name to "table-header-drag-drop", as the class is all about dragging only headers

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

**Review by msankaran0712 — APPROVED** [review]

**Pull request merged**